moped-editor/src/views/projects/projectView/ProjectSummary/SubprojectsTable.js

```diff
@@ -41,161 +134,171 @@ const SubprojectsTable = ({ projectId = null, refetchSummaryData }) => {
-            name="project_name"
-            options={data.subprojectOptions}
-            getOptionLabel={(option) =>
-              `${option.project_id} - ${option.project_name_full}`
-            }
-            value={props.value || null}
-            onChange={(event, value) => props.onChange(value)}
-            renderInput={(params) => (
-              <TextField variant="standard" {...params} />
-            )}
-          />
-          <FormHelperText>Required</FormHelperText>
-        </FormControl>
-      ),
+    [rowModesModel]
+  );
+
+  // handles canceling adding a new row by clicking the X icon
+  const handleCancelClick = useCallback(
+    (id) => () => {
+      setRowModesModel({
+        ...rowModesModel,
+        [id]: { mode: GridRowModes.View, ignoreModifications: true },
+      });
+      const editedRow = rows.find((row) => row.id === id);
+      if (editedRow.isNew) {
+        setRows(rows.filter((row) => row.id !== id));
+      }
     },
-    {
-      title: "Status",
-      field: "status",
-      editable: "never",
-      width: "25%",
-      customSort: (a, b) =>
-        a.moped_proj_phases?.[0]?.moped_phase?.phase_name <
-        b.moped_proj_phases?.[0]?.moped_phase?.phase_name
-          ? -1
-          : 1,
-      render: (entry) => (
-        <ProjectStatusBadge
-          phaseName={entry.moped_proj_phases?.[0]?.moped_phase?.phase_name}
-          phaseKey={entry.moped_proj_phases?.[0]?.moped_phase?.phase_key}
-          condensed
-        />
-      ),
+    [rowModesModel, rows]
+  );
+
+  // open the delete confirmation modal
+  const handleDeleteOpen = useCallback((id) => {
+    setIsDeleteConfirmationOpen(true);
+    setDeleteConfirmationId(id);
+  }, []);
+
+  // handles row delete
+  const handleDeleteClick = (id) => () => {
+    const childProjectId = id;
+    // remove row from rows in state
+    setRows(rows.filter((row) => row.project_id !== id));
+
+    deleteProjectSubproject({
+      variables: {
+        childProjectId: childProjectId,
+      },
+    })
+      .then(() => {
+        refetch();
+        refetchSummaryData(); // Refresh subprojects in summary map
+      })
+      .then(() => setIsDeleteConfirmationOpen(false))
+      .catch((error) => console.error(error));
+  };
+
+  const handleRowModesModelChange = (newRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
+
+  const handleProcessUpdateError = (error) => {
+    console.error(error.message);
+  };
+
+  // handles insert mutation triggered by row mode switching from edit to view
+  const processRowUpdate = useCallback(
+    (newRow) => {
+      const childProjectId = newRow?.project_name_full?.project_id;
+      // newRow.project_name_full is an object containing the id, name, and phase for the project selected by the
+      // lookup component, we will use it for setting all of the newRow fields needed to update the datagrid internal state
+      newRow.project_id = newRow.project_name_full.project_id;
```

Comment:
if i am using the autocomplete drop down and press `enter` without any dropdown item selected, the console errors:

```
Cannot read properties of null (reading 'project_id') 
```

it seems innocuous, but i wondered if you could mute it by making sure `project_name_full` is not null here 🤷